src/canopy_alloc.F90

```diff
@@ -46,16 +46,33 @@ SUBROUTINE canopy_alloc
     if(.not.allocated(lad))                allocate(lad(nlat*nlon,modlays))
     if(.not.allocated(lad_3d))             allocate(lad_3d(nlon,nlat,modlays))
 
-    if(.not.allocated(tleaf_sun24_tmp))    allocate(tleaf_sun24_tmp(ntime,nlat*nlon,modlays))
-    if(.not.allocated(tleaf_shade24_tmp))  allocate(tleaf_shade24_tmp(ntime,nlat*nlon,modlays))
-    if(.not.allocated(tleaf_ave24_tmp))    allocate(tleaf_ave24_tmp(ntime,nlat*nlon,modlays))
-    if(.not.allocated(ppfd_sun24_tmp))     allocate(ppfd_sun24_tmp(ntime,nlat*nlon,modlays))
-    if(.not.allocated(ppfd_shade24_tmp))   allocate(ppfd_shade24_tmp(ntime,nlat*nlon,modlays))
-    if(.not.allocated(tleaf_sun240_tmp))   allocate(tleaf_sun240_tmp(ntime,nlat*nlon,modlays))
-    if(.not.allocated(tleaf_shade240_tmp)) allocate(tleaf_shade240_tmp(ntime,nlat*nlon,modlays))
-    if(.not.allocated(tleaf_ave240_tmp))   allocate(tleaf_ave240_tmp(ntime,nlat*nlon,modlays))
-    if(.not.allocated(ppfd_sun240_tmp))    allocate(ppfd_sun240_tmp(ntime,nlat*nlon,modlays))
-    if(.not.allocated(ppfd_shade240_tmp))  allocate(ppfd_shade240_tmp(ntime,nlat*nlon,modlays))
+    if (hist_opt .eq. 1) then
+        if(.not.allocated(tleaf_sun24_tmp))    allocate(tleaf_sun24_tmp(25,nlat*nlon,modlays))
+        if(.not.allocated(tleaf_shade24_tmp))  allocate(tleaf_shade24_tmp(25,nlat*nlon,modlays))
+        if(.not.allocated(tleaf_ave24_tmp))    allocate(tleaf_ave24_tmp(25,nlat*nlon,modlays))
+        if(.not.allocated(ppfd_sun24_tmp))     allocate(ppfd_sun24_tmp(25,nlat*nlon,modlays))
+        if(.not.allocated(ppfd_shade24_tmp))   allocate(ppfd_shade24_tmp(25,nlat*nlon,modlays))
+        if(.not.allocated(tleaf_sun240_tmp))   allocate(tleaf_sun240_tmp(241,nlat*nlon,modlays))
+        if(.not.allocated(tleaf_shade240_tmp)) allocate(tleaf_shade240_tmp(241,nlat*nlon,modlays))
+        if(.not.allocated(tleaf_ave240_tmp))   allocate(tleaf_ave240_tmp(241,nlat*nlon,modlays))
+        if(.not.allocated(ppfd_sun240_tmp))    allocate(ppfd_sun240_tmp(241,nlat*nlon,modlays))
+        if(.not.allocated(ppfd_shade240_tmp))  allocate(ppfd_shade240_tmp(241,nlat*nlon,modlays))
+        if(.not.allocated(tmp2mref_tmp))       allocate(tmp2mref_tmp(25,nlat*nlon))
+        if(.not.allocated(ubzref_tmp))         allocate(ubzref_tmp(25,nlat*nlon))
+        if(.not.allocated(tleaf_sun24_rst))    allocate(tleaf_sun24_rst(24,nlat*nlon,modlays))
```

Comment:
Not sure we need the "rst" variables, please see my general comment on idea to output the total 24-hr and 240-hr leaf temp, ppfd in i,j,k (and maybe lai) as restart variables to be read in separately and adjusted to instantaneous based on fractions until new 24-hr and 240-hr become available.